workitems/l2-transaction-fees/l2-transaction-fees-v1.0-psd01.md

```diff
@@ -196,18 +260,270 @@ Note that requirements are uniquely numbered in ascending order within each requ
 
 -------
 
-# 3 Layer 2 Transaction Fees Specification
+# 3 Layer 2 Transaction Fee Specification
+
+In this section we will formulate requirements in the following areas:
+- Transparency
+- Visibility and Roles
+- Transactions
+
+## 3.1 Layer 2 Transaction Fee Transparency Requirements
+
+#### **[R1]**
+
+A L2 Transaction Fee MUST be comprised as the sum of a Base Fee, a Execution Fee, and a Priority Fee.
+
+[[R1]](#r1) Testability: Mathematical equality equations can be translated into computer code, and any computer code is testable. Therefore, the requirement is testable.
+
+#### **[R2]**
+
+All components of a L2 Transaction Fee MUST NOT be less than zero.
+
+[[R2]](#r2) Testability: Mathematical equality equations can be translated into computer code, and any computer code is testable for specific conditions such as the sum of three elements, or each individual element being less than zero. Therefore, the requirement is testable.
+
+#### **[O1]**
+
+A L2 Transaction Fee or any of its components MAY BE zero.
+
+[[O1]](#o1) Testability: Mathematical equality equations can be translated into computer code, and any computer code is testable for specific conditions such as the sum of three elements, or each individual element being zero. Therefore, the requirement is testable.
+
+#### **[R3]**
+
+The setting and/or calculation of a Base Fee MUST be well documented and verifiable.
+
+[[R3]](#r3) Testability: Documentation about the setting and/or calculation of a Base Fee can be reviewed by individuals, and subsequently compared to computer code that sets and/or calculates a Base Fee. And computer code is always testable. Therefore, the requirement is testable.
+
+#### **[R4]**
+
+The setting and/or calculation of an Execution Fee MUST be well documented and verifiable.
+
+[[R4]](#r4) Testability: Documentation about the setting and/or calculation of an Execution can be reviewed by individuals, and subsequently compared to computer code that sets and/or calculates an Execution Fee. And computer code is always testable. Therefore, the requirement is testable.
+
+#### **[R5]**
+
+The setting of a Priority Fee MUST be well documented and verifiable.
+
+[[R5]](#r5) Testability: Documentation about the setting of a Priority Fee can be reviewed by individuals, and subsequently compared to computer code that sets a Priority Fee. And computer code is always testable. Therefore, the requirement is testable.
+
+#### **[R6]**
+
+A L2 MUST provide a capability to estimate a Transaction Fee based on a given Transaction and the current operating state of the L2.
+
+Note that L2 operating state refers to a combination of transaction volume being processed on an L2 and the state of all Accounts on an L2.
```

Comment:
Hello, was curious on the 2nd part of the sentence:
"the state of all Accounts on an L2."
Could you please elaborate on this? Not really sure what is being meant here.

Reply:
@0x000hx sure. This is referring to how many accounts are being operated on at any given time. to simplify it, operating state = all transactions in the mempool plus all transactions currently processing.

Reply:
My concern is that the word "state" is being used in a specific meaning in our industry, so maybe "load conditions" would work better?

Reply:
Seems good, thanks!